### ISSUE TITLE
Scheduled event deletion though DELETE /recordings/ REST API broken

### ISSUE THREAD

**Issue by Rillke** [opencast/opencast#4273]
**Describe the bug**
A clear and concise description of what you see:
*When scheduling an event and later deleting it through **`DELETE /recordings/EVENT_ID`** REST endpoint, the event remains without schedule and deletion on the admin UI throws an error on the admin UI after which the event disappears from the admin UI*

**To Reproduce**
1. Go to '[stable.opencast.org](https://stable.opencast.org/admin-ng/index.html)' and login as admin.
2. Click on '+ Add event'  <br><img width="163" alt="Screenshot 2022-10-04 at 15 18 55" src="https://user-images.githubusercontent.com/2311611/193829606-8b2a8725-0376-4982-a404-6cf8b2f58211.png">
3. Fill a title data, click next.
4. Schedule an event at any location available, select any workflow, do not change the ACLs suggested and create the event.
5. Click the event <img width="38" alt="Screenshot 2022-10-04 at 15 22 09" src="https://user-images.githubusercontent.com/2311611/193830346-c0ddfcad-c518-43b5-a791-60fac1f34c0b.png"> details button in order to find the event's UUID. 
6. Go to https://stable.opencast.org/docs.html?path=/recordings#deleterecordings-1 (? -> REST-API-Docs -> /recordings -> DELETE)
7. Enter the UUID of the event you just created and click "submit"<br><img width="1036" alt="Screenshot 2022-10-04 at 15 22 54" src="https://user-images.githubusercontent.com/2311611/193830559-8240ee46-0d17-4b4f-81e0-d3c6e512943c.png">
8. **Result: Scheduling itab s gone in admin UI and the status switched from scheduled to finished. Expected: Event to be gone entirely as in older versions of Opencast.**
9. Click the red "delete" mark <img width="36" alt="Screenshot 2022-10-04 at 15 24 13" src="https://user-images.githubusercontent.com/2311611/193830889-1142975c-4df1-4890-b4bd-e870e16d022c.png"> next to the event and confirm.
10. See error message: "The event(s) could not be deleted."

Log:

```
2022-10-04T15:25:15,272 | INFO  | (IngestServiceImpl:632) - Created mediapackage bf6f4e87-3c42-44e1-9e2a-37e217d049b1
2022-10-04T15:25:15,307 | INFO  | (AssetManagerImpl:1262) - Creating new version 0 of media package bf6f4e87-3c42-44e1-9e2a-37e217d049b1
2022-10-04T15:25:15,316 | INFO  | (AssetManagerImpl:413) - Trigger update handlers for snapshot bf6f4e87-3c42-44e1-9e2a-37e217d049b1, version 0
2022-10-04T15:25:23,766 | WARN  | (SchedulerServiceImpl:1518) - Scheduled recording  bf6f4e87-3c42-44e1-9e2a-37e217d049b1 not found for deletion from the Elasticsearch index.
2022-10-04T15:25:23,766 | INFO  | (SchedulerRestService:506) - Event with id ' bf6f4e87-3c42-44e1-9e2a-37e217d049b1' does not exist.
2022-10-04T15:25:30,503 | INFO  | (AssetManagerImpl:870) - Firing event handlers for event bf6f4e87-3c42-44e1-9e2a-37e217d049b1, snapshot 0
2022-10-04T15:25:30,503 | INFO  | (AssetManagerImpl:876) - Firing event handlers for event bf6f4e87-3c42-44e1-9e2a-37e217d049b1
```

Screencast:

https://user-images.githubusercontent.com/2311611/193832645-7462f273-4b42-44f5-8904-70b197562994.mp4

XHR (deletion from admin UI):

<img width="1309" alt="Screenshot 2022-10-04 at 15 33 47" src="https://user-images.githubusercontent.com/2311611/193833182-7fddc2bc-29d4-419a-a890-f1025b6489c5.png">
Response: 3e42c43c-af5f-4c68-b4dd-4c0908ec1325

**Environment Information**
 - Browser Firefox
 - Opencast version with minor release 12.3

**Comment by CGreweling** [comment]
@Rillke take a look at #4084

**Comment by mliradelc** [comment]
I just made some tests with
`DELETE /admin-ng/event/{eventID}`
`DELETE /api/events/{eventID}`

Both worked on [stable build](stable.opencast.org)


#4084 was from using those API entry points. But in your case, this bug is with `/recordings` API

**Comment by Rillke** [comment]
@mliradelc

> https://github.com/opencast/opencast/issues/4084 was from using those API entry points. But in your case, this bug is with /recordings API

Indeed. The issue is that this API is used by the Stud.IP Opencast Plugin and the second issue is that the REST documentation says

> Removes scheduled event with specified ID.

Therefore, it's not entirely clear to me, if the 12.3 behavior might be correct or wanted. You could interpret it the following way: Opencast removes the scheduled event but leaves behind the event without scheduling information. However, if this is the wanted behavior and not a bug, we need to add this to the release notes.

**Comment by mliradelc** [comment]
True, I think this should be talk today on the developers meeting, because, what is the meaning of `DELETE /recordings` if the other two endpoints resolves the same (One for the admin-ui only, and the another for any other external that needs to communicate with opencast).

**Comment by Rillke** [comment]
Quoting @gregorydlogan "if an endpoint creates the event, its delete endpoint should also remove it". Anyone else?

**Comment by Alaschinger** [comment]
I just looked over the code and it seems like that is a wanted behaviour, even if I don't know why. It could easily be fixed with #4369 by deleting the event. Currently there is an update function which removes the capture agent from the index entry and leaves the event as it is.

**Comment by JulianKniephoff** [comment]
This was resolved by #4273, right? It's not automatically closed because that PR only mentioned this issue in the description and was not targeting our default branch.

Of course anyone is invited to reopen this if it is still an issue. 😄
